discord_bots/commands.py

```diff
@@ -3986,160 +4012,167 @@ def win_rate(wins, losses, ties):
 async def stats(ctx: Context):
     player_id = ctx.message.author.id
     session = Session()
-    fgps = (
-        session.query(FinishedGamePlayer)
-        .filter(FinishedGamePlayer.player_id == player_id)
-        .all()
-    )
-    finished_game_ids = [fgp.finished_game_id for fgp in fgps]
-    fgs = (
-        session.query(FinishedGame).filter(FinishedGame.id.in_(finished_game_ids)).all()
-    )
-    fgps_by_finished_game_id: dict[str, FinishedGamePlayer] = {
-        fgp.finished_game_id: fgp for fgp in fgps
-    }
-
     player: Player = session.query(Player).filter(Player.id == player_id).first()
-    players: list[Player] = session.query(Player).all()
+    channel_output = ""
 
-    default_rating = Rating()
-    DEFAULT_TRUESKILL_MU = float(os.getenv("DEFAULT_TRUESKILL_MU") or default_rating.mu)
-
-    # Filter players that haven't played a game
-    players = list(
-        filter(
-            lambda x: x.rated_trueskill_mu != default_rating.mu
-            and x.rated_trueskill_mu != DEFAULT_TRUESKILL_MU,
-            players,
+    if not player.stats_enabled:
+        channel_output = "You have disabled !stats"
```

Comment:
better here i think is to send the channel message and return early, then the rest of the function is unchanged. so like 

```
    session.close()
    if ctx.message.guild:
        member_: Member | None = ctx.message.guild.get_member(player.id)
        await send_message(
            ctx.message.channel,
            embed_description=channel_output,
            colour=Colour.blue(),
        )
        return
```

Reply:
yeah, i'm always trying to DRY as much as possible, but having such a huge else block felt bad too